test/test.js

```diff
@@ -10,6 +10,12 @@ function sortLeftToRight(a, b) {
 	return 0;
 }
 
+function assertTree(tree, nbElements) {
+	assert.strictEqual(tree._getCount(tree.root), nbElements);
+	assert.strictEqual(tree._isSorted(tree.root), true);
```

Comment:
As you can see here, code is not verifiably correct.  
This fails all my tests, and passes all yours. 
https://github.com/Wizcorp/container-avltree/pull/5
Tests are there to help the developer verify correctness, and avoid bugs, when adding features, fixing bugs, or refactoring.  Your tests only protect against 1? issue.  There was no reason to delete my tests, as they all pass with this code.  If they failed while you were making your optimizations, then they did their job.